### DISCUSSION TITLE
About resource loading timeout

### DISCUSSION THREAD

**Discussion by w1ndys** [squidfunk/mkdocs-material#6918] [Q&A] [answered]
My main users are users in China, and my deployment platform is on Vercel. Due to the Great Firewall, the speed for our Chinese users to access the site is not very fast. I found out during background debugging that this resource comes from Google, which is even slower. In order to improve the loading speed of the page, is there a way to speed it up (other than VPN), such as saving the resource of this link locally, or other better methods?

![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/e0824988-3479-4ef7-a0b0-8feb3591377b)

The request URL mentioned on the page is 
```
https://fonts.googleapis.com/css?family=Noto+Sans+SC:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback
```

And
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/94c3b8cb-83e6-4a91-9cbb-bf636e3d3490)

Is this because of the connection problem to Vercel or the connection problem to the resource address? I don’t understand where the problem lies.

I checked and this is also Google's public CDN

I tried to use privacy, but an error occurred during Vercel deployment

![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/49e423d4-8c7a-439a-85fb-43d015c1a05b)

**Comment by alexvoss** [comment, marked as answer]
For the fonts, have a look at the [changing the fonts setup guide](https://squidfunk.github.io/mkdocs-material/setup/changing-the-fonts/), which shows how to avoid loading fonts from Google. You may want to check if that helps. 

The [privacy plugin](https://squidfunk.github.io/mkdocs-material/plugins/privacy/) might help you automate the process of turning all resources that come from a CDN into locally served resources. Ah, I see you just tried that. Let me see if I can find out what the issue might be with the error you are seeing.

**Reply by alexvoss** [reply]
This thread might help with the issue you are experiencing: https://github.com/squidfunk/mkdocs-material/discussions/6470

**Reply by squidfunk** [reply]
Correct, the privacy plugin is the way to go! It allows you to fetch resources _once_ at build time, and serve all of them via one IP, omitting requests to Google. I think we should mention this somewhere on the documentation @alexvoss @katharinalisalin

**Reply by alexvoss** [reply]
Happy to suggest something but we might want to think about common use cases other than the Great Firewall or other such general connectivity problems? Internally hosted (versions of) documentation perhaps?

**Comment by squidfunk** [comment]
The SSL problem should be raised to Vercel. [OpenSSL 1.0.2 is from January 2015 – 9 years old](https://en.wikipedia.org/wiki/OpenSSL#Major_version_releases). I bet there are dozens of CVEs.

**Comment by w1ndys** [comment]
@squidfunk @alexvoss 
You guys are awesome, I have solved this problem
I only need to control the versions of the involved software packages in requirements.txt to avoid SSL problems. I have seen the effect in the deployment log.
```
urllib3<2
```
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/59846221-1aee-48a0-ba60-100772de18a8)
